internal/tofu/node_resource_apply_instance.go

```diff
@@ -439,7 +437,6 @@ func (n *NodeApplyableResourceInstance) managedResourceExecute(ctx context.Conte
 	}
 
 	diags = diags.Append(n.postApplyHook(evalCtx, state, diags.Err()))
-	diags = diags.Append(updateStateHook(evalCtx))
```

Comment:
Just to be sure: the call above to `MaybeRestoreResourceInstanceDeposed` is meant to store that deposed information only in the EvaluateContext that is meant to be used later and not meant to be stored in the state snapshot, right?
Want to be sure that by removing this `updateState` from here will have no side effects.

Reply:
Good catch, let me restore this.